aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -946,12 +979,11 @@ module aptos_framework::stake {
         // Reset performance scores and update active validator set (so network address/public key change takes effect).
         let i = 0;
         let validator_index = 0;
-        let len = vector::length(&validator_set.active_validators);
+        let vlen = vector::length(&validator_set.active_validators);
         let next_epoch_validators = vector::empty();
         validator_perf.validators = vector::empty();
-        validator_set.total_voting_power = 0;
-        validator_set.total_joining_power = 0;
-        while (i < len) {
+        let total_voting_power = 0;
```

Comment:
This temp variable is introduced to make `validator_set` stay unchanged during the loop and make the verification easier.